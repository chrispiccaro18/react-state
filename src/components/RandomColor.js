import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from '../index.css';

export default class RandomColor extends PureComponent {
  static propTypes = {
    colors: PropTypes.array.isRequired
  }

  state = {
    color: 'blue'
  }

  componentDidMount() {
    this.timerId = setInterval(
      () => this.randomColorGen(),
      1000
    );
  }

  componentWillUnmount() {
    clearInterval(this.timerId);
  }

  randomColorGen() {
    const random = this.props.colors[Math.floor(Math.random() * this.props.colors.length)];
    if(random === this.state.color) {
      this.setState({ color: 'duplicate' });
    } else {
      this.setState({ color: random });
    }
  }
  
  render() {
    return (
      <div className={styles[this.state.color]}></div>
    );
  }
}
