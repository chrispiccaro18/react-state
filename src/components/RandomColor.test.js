import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from './RandomColor';

describe('RandomColor component', () => {
  it('renders a RandomColor', () => {
    const wrapper = shallow(<RandomColor colors={['red', 'yellow', 'blue']}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
