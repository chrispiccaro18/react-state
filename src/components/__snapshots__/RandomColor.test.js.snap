// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomColor component renders a RandomColor 1`] = `
<div
  className="blue"
/>
`;
